src/shogun/regression/GLM.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Ahmed Khalifa
+ */
+/*
+References:
+https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/family
+https://www.statsmodels.org/stable/generated/statsmodels.genmod.generalized_linear_model.GLM.html#statsmodels.genmod.generalized_linear_model.GLM
+http://glm-tools.github.io/pyglmnet/api.html
+*/
+#ifndef _GENERALIZEDLINEARMODEL_H__
+#define _GENERALIZEDLINEARMODEL_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/features/Features.h>
+#include <shogun/machine/FeatureDispatchCRTP.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/optimization/DescendUpdater.h>
+#include <shogun/regression/Regression.h>
+
+namespace shogun
+{
+	enum Family
+	{
+		NORMAL_DISTRIBUTION,
+		EXPONENTIAL_DISTRIBUTION,
+		GAMMA_DISTRIBUTION,
+		BINOMIAL_DISTRIBUTION,
+		GAUSS_DISTRIBUTION,
+		POISSON_DISTRIBUTION
+	};
+	enum LinkFunction
+	{
+		LOG,
+		LOGIT,
+		IDENTITY,
+		INVERSE
+	};
+	/** @brief Class GLM implements Generalized Linear Models, such as poisson,
+	 * gamma, binomial
+	 */
+	class GLM : public LinearMachine
+	{
+	public:
+		/** problem type */
+		MACHINE_PROBLEM_TYPE(PT_REGRESSION);
+
+		/** Default constructor */
+		GLM();
+
+		/** Constructor
+		 *
+		 * @param descend_updater chosen Descend Updater algorithm
+		 * @param Linkfn the link function check
+		 * https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/family
+		 * @param Family the family check
+		 * https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/family
+		 * @param alpha alpha
```

Comment:
what are alpha and lambda? could you also describe them in the docs if you put docs? (otherwise they are pointless)